consensus/types/src/chain_spec.rs

```diff
@@ -225,14 +225,17 @@ impl ChainSpec {
     /// Returns a full `Fork` struct for a given epoch.
     pub fn fork_at_epoch(&self, epoch: Epoch) -> Fork {
         let current_fork_name = self.fork_name_at_epoch(epoch);
-        let previous_fork_name = current_fork_name.previous_fork().unwrap_or(ForkName::Base);
-        let epoch = self
-            .fork_epoch(current_fork_name)
-            .unwrap_or_else(|| Epoch::new(0));
+        self.fork_for_name(current_fork_name)
+    }
+
+    /// Returns a full `Fork` struct for a given `ForkName`.
+    pub fn fork_for_name(&self, fork_name: ForkName) -> Fork {
+        let previous_fork_name = fork_name.previous_fork().unwrap_or(ForkName::Base);
+        let epoch = self.fork_epoch(fork_name).unwrap_or_else(|| Epoch::new(0));
```

Comment:
I don't think we should default to the zero epoch here because it will cause it to return bogus `Fork` structs for forks that never activate. Instead I think we should return an `Option<Fork>` and return `None` in the case where the fork never activates. This might require duplicating some of the logic from `fork_at_epoch` (which can continue to return `Fork`).

This will be particularly relevant once @pawanjay176 changes our config reader to convert the spec's `u64::MAX` into a `None` epoch as per https://github.com/sigp/lighthouse/pull/2532

Reply:
That's a good point, it's pretty weak to just rely on `ForkName` variants.

Reply:
Updated :+1: